cirq/devices/line_qubit.py

```diff
@@ -184,6 +186,8 @@ class LineQubit(_BaseLineQid):
         cirq.LineQubit(1)
     """
 
+    x: int
```

Comment:
I don't think this is right considering the inheritance here.

Reply:
I think all three classes here should be dataclasses with `_BaseLineQid` defining the `x` field and `LineQid` defining the `dimension` field.